src/main/kotlin/com/exactpro/th2/ws/client/api/impl/WebSocketClient.kt

```diff
@@ -42,28 +42,32 @@ class WebSocketClient(
     private val textFrames = mutableListOf<String>()
     private val binaryFrames = mutableListOf<ByteArray>()
     private val lock = ReentrantLock()
-    private lateinit var socket: WebSocket
+    @Volatile private lateinit var socket: WebSocket
 
     @Volatile var isRunning: Boolean = false
         private set
 
     private fun awaitSocket(): WebSocket {
-        when {
-            !isRunning -> start()
-            !::socket.isInitialized || socket.isOutputClosed -> connect()
+        if (!isRunning) start()
+
+        while (!::socket.isInitialized || socket.isOutputClosed) {
+            when {
+                isRunning -> Thread.sleep(1)
+                else -> return awaitSocket()
+            }
         }
 
         return socket
     }
 
-    override fun sendText(text: String) = lock.withLock {
+    override fun sendText(text: String) {
```

Comment:
Why do we keep using the lock in `sendPing` method?

Reply:
I just forgot about it. Thanks